src/System.IO.FileSystem/tests/FileSystemTest.cs

```diff
@@ -49,5 +49,51 @@ public abstract partial class FileSystemTest : RemoteExecutorTestBase
 
             return success;
         });
+
+        /// <summary>
+        /// Runs the given command as sudo
+        /// </summary>
+        /// <param name="commandLine">The command line to run as sudo</param>
+        /// <returns> Returns 0 if the process exits</returns>
+        protected static int RunAsSudo(string commandLine)
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo()
+            {
+                FileName = "sudo",
+                Arguments = commandLine
+            };
+
+            using (Process process = Process.Start(startInfo))
+            {
+                process.WaitForExit();
+                return process.ExitCode;
+            }
+        }
+
+        protected void ReadOnly_FileSystemHelper(Action<string> testAction)
+        {
+            // Set up read only file system
+            // Set up the source directory
+            string sourceDirectory = GetTestFilePath();
+            string sourceSubDirectory = Path.Combine(sourceDirectory, "subdir");
```

Comment:
The tests that use this helper need to know what this magic string is.  It'd be better if the subdir string was passed into the helper, so that the calling tests can just tell the helper what name to use rather than having to ensure they match in multiple source locations.